### PR TITLE
[ide-metrics] add error reporting api

### LINKED ISSUE #11915: IDE clients logs
After https://github.com/gitpod-io/gitpod/pull/11542 we have an API endpoint which allow to collect IDE metrics. We should enrich it to allow collection of IDE client error logs and reporting them to [GCP error logging](https://cloud.google.com/error-reporting).

### PR DESCRIPTION
**Pull request by iQQBot**
## Description

This PR enrich `ide-metrics` to allow collection of IDE client error logs and reporting them to GCP error logging.

On `gitpod.io` we reporting them to GCP, on `preview environment` we write it in `stdin`, on `self-hosted` we ignore it.

## Related Issue(s)

Fixes #11915

## How to test

1. run below curl command
```
## cURL
curl -X "POST" "https://ide.pd-logging-api.preview.gitpod-dev.com/metrics-api/reportError" \
     -H 'Content-Type: application/json; charset=utf-8' \
     -d $'{
  "component": "supervisor-frontend",
  "workspaceId": "workspaceId1",
  "error_stack": "Error: test-error\\n    at REPL16:1:7\\n    at Script.runInThisContext (node:vm:129:12)\\n    at REPLServer.defaultEval (node:repl:566:29)\\n    at bound (node:domain:421:15)\\n    at REPLServer.runBound [as eval] (node:domain:432:12)\\n    at REPLServer.onLine (node:repl:893:10)\\n    at REPLServer.emit (node:events:539:35)\\n    at REPLServer.emit (node:domain:475:12)\\n    at REPLServer.[_onLine] [as _onLine] (node:internal/readline/interface:418:12)\\n    at REPLServer.[_line] [as _line] (node:internal/readline/interface:879:18)",
  "instanceId": "instanceId1",
  "userId": "userId",
  "properties": {
    "p2": "v2",
    "p1": "v1"
  }
}'
```
2. see pod `ide-metrics` logs, it should be log.
3. try to modify `component` to another value and execute again, it should be reject because component has an allowlist.

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

File: components/ide-metrics/pkg/errorreporter/error_reporter.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	})
+}
+
+type logErrorReporter struct {
+}
+
+func (r *logErrorReporter) Report(event ReportedErrorEvent) {
+	b, err := json.Marshal(event)
+	if err != nil {
+		log.WithError(err).Error("cannot marshal error")
```
Comment (andreafalzetti) [inline components/ide-metrics/pkg/errorreporter/error_reporter.go:75]: Should we log the `event` to make debugging easier?

**Comment by jeanp413** [comment]
I think these endpoints needs authentication so they're not abused so easily

**Comment by mustard-mh** [comment]
Tested well as promised

image

**Comment by akosyakov** [comment]
> I think these endpoints needs authentication so they're not abused so easily

It is the same as with analytics or observability. We are going to add rate limitting per ip to prevent abuse. Or maybe we already have. The thing is that we did not have problems so far. So I think it should be done separately.

**Review by mustard-mh — APPROVED** [review]
Nice!

**Review by akosyakov — APPROVED** [review]
lgtm, let's bring it 🚀

**Review by nandajavarma — APPROVED** [review]

**Pull request merged**